packages/api/pages/api/languages/[code]/verses/[verseId]/glosses.ts

```diff
@@ -1,13 +1,78 @@
 import { GetVerseGlossesResponseBody } from '@translation/api-types';
 import { client, PrismaTypes } from '../../../../../../shared/db';
 import createRoute from '../../../../../../shared/Route';
+import { machineTranslationClient } from '../../../../../../shared/machine-translation';
+import languageMap from '../../../../../../../../data/language-mapping.json';
+import { Language, Prisma } from '@prisma/client';
 
-type WordsRawQuery = {
+interface WordQuery {
   wordId: string;
-  gloss: string;
+  gloss?: string;
   suggestions: string[];
   state: PrismaTypes.GlossState;
-}[];
+  refGloss?: string;
+  machineGloss?: string;
+}
+
+// This is an optimized sql command for saving the machine glosses for future use.
+// The goal is to save the gloss produced by Google Translate for every word in the database that has the same English gloss.
```

Comment:
Why is it based on the English gloss, instead of the original language word?

Reply:
Because Google can't translate from the original languages. We translate from the english glosses, so we can update every word in the database that has the same English gloss regardless of the original language word.